### PR TITLE
Support getting browser version from firefox_android

### PR DESCRIPTION
**Pull request by jgraham**
Initially we try to use mozversion on the package, but if that doesn't work we try to get the version from adb dumpsys.

### CODE REVIEW

File: infrastructure/metadata/infrastructure/server/webtransport-h3.https.sub.any.js.ini
Comment (jcscottiii) [inline infrastructure/metadata/infrastructure/server/webtransport-h3.https.sub.any.js.ini:1]: I'm assuming the changes to this file are related to the webtransport issue you are going to file. Is this change needed for this PR though?

File: tools/wptrunner/wptrunner/browsers/firefox_android.py [hunk 1]
```diff
@@ -87,14 +90,45 @@ def env_extras(**kwargs):
...
+        if rv.get("browser_version") is None:
+            # If we didn't get the browser version from the apk, try to get it from adb dumpsys
+            rv["browser_version"] = get_package_browser_version(logger,
+                                                                kwargs["adb_binary"],
+                                                                kwargs["package_name"])
+
     return rv
 
 
+def get_package_browser_version(logger, adb_binary, package_name):
```
Comment (jcscottiii) [inline tools/wptrunner/wptrunner/browsers/firefox_android.py:111]: Is there a reason not to use the same logic from here?

https://github.com/web-platform-tests/wpt/blob/d153eb36f37e71f3b64017952d6cc6d88724ed54/tools/wpt/browser.py#L1452-L1471

I imagine there could be a new abstract class called AndroidBase `class AndroidBase(BrowserSetup)`. AndroidBase has the `version` method implemented (by moving the implementation from ChromeAndroidBase to AndroidBase) .

Both ChromeAndroidBase and FirefoxAndroid are child classes of this new class. 

That way we don't have to repeat the same code or have a different pattern for getting the version.

**Review by jcscottiii — COMMENTED** [review]
@jgraham one question about using existing code and pattern to get the version.

File: tools/wpt/run.py [hunk 2]
```diff
@@ -395,17 +395,18 @@ def setup_kwargs(self, kwargs):
                               device_serial=device_serial,
                               prompt=kwargs["prompt"])
 
-        if "ADB_PATH" not in os.environ:
-            adb_path = os.path.join(android.get_paths(None)["sdk"],
-                                    "platform-tools",
-                                    "adb")
-            os.environ["ADB_PATH"] = adb_path
-        adb_path = os.environ["ADB_PATH"]
+        if kwargs["adb_binary"] is None:
```
Comment (whimboo) [inline tools/wpt/run.py:398]: Is `adb_binary` always set? If not we probably want to use `kwargs.get()` here.

File: tools/wptrunner/wptrunner/browsers/firefox_android.py [hunk 3]
```diff
@@ -87,14 +90,45 @@ def env_extras(**kwargs):
...
+    if adb_binary is None:
+        logger.warning("Couldn't run adb to get Firefox Android version number")
+        return None
+    try:
+        completed = subprocess.run([adb_binary, "shell", "dumpsys", "package", package_name],
+                                   check=True,
+                                   capture_output=True,
+                                   encoding="utf8")
+    except subprocess.CalledProcessError as e:
+        logger.warning(f"adb failed with return code {e.returncode}\nCaputued stderr:\n{e.stderr}")
```
Comment (whimboo) [inline tools/wptrunner/wptrunner/browsers/firefox_android.py:121]:
```suggestion
        logger.warning(f"adb failed with return code {e.returncode}\nCaptured stderr:\n{e.stderr}")
```

File: tools/wptrunner/wptrunner/executors/executormarionette.py [hunk 4]
```diff
@@ -1037,8 +1037,13 @@ def __init__(self, logger, browser, server_config, timeout_multiplier=1,
         self.implementation_kwargs = {}
         if reftest_internal:
             self.implementation_kwargs["screenshot"] = reftest_screenshot
-            self.implementation_kwargs["chrome_scope"] = (browser_version is not None and
-                                                          int(browser_version.split(".")[0]) < 82)
+            self.implementation_kwargs["chrome_scope"] = False
+            if browser_version is not None:
+                try:
+                    major_version = int(browser_version.split(".")[0])
+                    self.implementation_kwargs["chrome_scope"] = major_version < 82
```
Comment (whimboo) [inline tools/wptrunner/wptrunner/executors/executormarionette.py:1044]: Hm, what exactly is this check for? Maybe you can add a comment?

File: infrastructure/metadata/infrastructure/server/webtransport-h3.https.sub.any.js.ini
Comment (jgraham) [reply infrastructure/metadata/infrastructure/server/webtransport-h3.https.sub.any.js.ini:1]: Apparently to make it pass CI. But I don't really understand why, given that this PR isn't touching webtransport. However since the change is just "these tests no longer timeout, just fail" it seemed safe enough.

File: tools/wpt/run.py (hunk 2 above)
Comment (jgraham) [reply tools/wpt/run.py:398]: Yes, it comes from the command line args and we give everything a default value.

File: tools/wptrunner/wptrunner/browsers/firefox_android.py (hunk 1 above)
Comment (jgraham) [reply tools/wptrunner/wptrunner/browsers/firefox_android.py:111]: Yes, so I don't want this in the wpt frontend because at least gecko CI doesn't use that frontend, and I'd prefer to keep this logic in the shared code, as it is for desktop Firefox.

Arguably we could move stuff out of the frontend for other browsers, but that seemed out of scope for this PR.

File: tools/wptrunner/wptrunner/executors/executormarionette.py (hunk 4 above)
Comment (jgraham) [reply tools/wptrunner/wptrunner/executors/executormarionette.py:1044]: Note that this check already existed; I just made it more robust.

I think the origin is that in older gecko versions you had to switch to chrome scope to use the marionette reftest commands, but I'm not totally sure. _Probably_ we should just delete the check entirely now.

**Review by jcscottiii — APPROVED** [review]

**Review by whimboo — APPROVED** [review]
